sensor/sensor_calibration_manager/sensor_calibration_manager/calibrators/x2/tag_based_pnp_calibrator.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Tier IV, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict
+
+import numpy as np
+from sensor_calibration_manager.calibrator_base import CalibratorBase
+from sensor_calibration_manager.calibrator_registry import CalibratorRegistry
+from sensor_calibration_manager.ros_interface import RosInterface
+from sensor_calibration_manager.types import FramePair
+
+
+@CalibratorRegistry.register_calibrator(
+    project_name="x2", calibrator_name="tag_based_pnp_calibrator"
+)
+class TagBasedPNPCalibrator(CalibratorBase):
+    required_frames = []
+
+    def __init__(self, ros_interface: RosInterface, **kwargs):
+        super().__init__(ros_interface)
+
+        self.camera_name = kwargs["camera_name"]
+        self.lidar_frame = kwargs["lidar_frame"]
+
+        camera_name_to_sensor_kit_frame = {
+            "camera0": "top_unit_base_link",
+            "camera1": "top_unit_base_link",
+            "camera2": "top_unit_base_link",
+            "camera3": "rear_unit_base_link",
+            "camera4": "top_unit_base_link",
+            "camera5": "top_unit_base_link",
+            "camera6": "front_unit_base_link",
+            "camera7": "top_unit_base_link",
+        }
+
+        self.sensor_kit_frame = camera_name_to_sensor_kit_frame[self.camera_name]
+
+        self.required_frames.extend(
+            [
+                self.lidar_frame,
+                self.sensor_kit_frame,
+                f"{self.camera_name}/camera_link",
+                f"{self.camera_name}/camera_optical_link",
+            ]
+        )
+
+        self.add_calibrator(
+            service_name="calibrate_camera_lidar",
+            expected_calibration_frames=[
+                FramePair(parent=f"{self.camera_name}/camera_optical_link", child=self.lidar_frame),
+            ],
+        )
+
+    def post_process(self, calibration_transforms: Dict[str, Dict[str, np.array]]):
+        camera_to_lidar_transform = calibration_transforms[
```

Comment:
Same issue

Reply:
Addressed in [45453cf](https://github.com/tier4/CalibrationTools/pull/148/commits/45453cfff40144d04f022c1b3632290b6d2ed8ca)

Reply:
Addressed in [45453cf](https://github.com/tier4/CalibrationTools/pull/148/commits/45453cfff40144d04f022c1b3632290b6d2ed8ca)